include/meltpooldg/heat/heat_transfer_operator.hpp

```diff
@@ -834,16 +1019,87 @@ namespace MeltPoolDG::Heat
 
   private:
     /*
-     * @return \c std::tuple with density, capacity and conductivity
+     * Determine material parameters (capacity, conductivity and density) for
+     * solidification/melting. In the mushy zone (between solidus and liquidus temperautre) the
+     * material parameters are linearly interpolated between the solid's and fluid's values.
+     *
+     * @TODO, handle variable fluid parameters. ATM the fluid's parameters are assumed to be
+     * materal.second.<..>
+     */
+    void
+    get_material_parameters_with_solidification(VectorizedArrayType &      capacity,
```

Comment:
Since you always call `get_material_parameters` by `get_material_parameters_derivatives` would it make sense to have only one function, delivering both -- parameters and derivatives?
```suggestion
    get_solidifying_material_parameters_and_derivatives(VectorizedArrayType &      capacity,
     VectorizedArrayType &      conductivity,
     VectorizedArrayType &      density,
      VectorizedArrayType &      d_capacity_dT,
      VectorizedArrayType &      d_conductivity_dT,
      VectorizedArrayType &      d_density_dT,
      const VectorizedArrayType &temperature) const
```

Reply:
I only call both for the tangent functions. For the residual we don't need the derivatives.

Reply:
Okay, then please ignore my previous comment :-).